app/src/main/java/com/example/evently/ui/model/EntrantEventsViewModel.java

```diff
@@ -0,0 +1,115 @@
+package com.example.evently.ui.model;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.google.firebase.Timestamp;
+
+import com.example.evently.data.EventsDB;
+import com.example.evently.data.generic.Promise;
+import com.example.evently.data.model.Category;
+import com.example.evently.data.model.Event;
+import com.example.evently.utils.FirebaseAuthUtils;
+
+/**
+ * Shared {@link ViewModel} for Entrant event lists and filtering state.
+ */
+public class EntrantEventsViewModel extends ViewModel {
```

Comment:
I don't think we need to filter joined events, only browse events.

This viewmodel should ideally be simpler if possible